sphinx_gallery/gen_rst.py

```diff
@@ -55,11 +55,25 @@ def prefixed_lines():
 
 from io import StringIO
 
+mpl_backend_msg = """
+
+Sphinx-Gallery relies on the matplotlib Agg backend to render figures
+and write them to files. You are currently using the {} backend.
+Sphinx-Gallery will terminate the build now, because changing backends
+is not well supported by matplotlib. We advise you to review your
+Sphinx setup to use the Agg backend.\n\n"""
```

Comment:
The recommendation should be "Move the sphinx_gallery imports at the top of your conf.py file" rather than "review your configuration", Maybe a more nuanced thing like "sphinx_gallery imports should be before any matplotlib-dependent import. Moving the sphinx_gallery imports at the top of your conf.py file should fix this issue" or something better phrased.